Sources/Customization/DefaultEventDispatcher.swift

```diff
@@ -272,31 +266,3 @@ extension DefaultEventDispatcher {
     }
     
 }
-
-// MARK: - Notification Observers
-
-extension DefaultEventDispatcher {
-    
-    func addProjectChangeNotificationObservers() {
-        observers.append(
-            NotificationCenter.default.addObserver(forName: .didReceiveOptimizelyProjectIdChange, object: nil, queue: nil) { [weak self] _ in
-                self?.logger.d("Event flush triggered by datafile projectId change")
-                self?.flushEvents()
```

Comment:
after removing this, the previous events will still be flushed before the new ones come in?

Reply:
The old events still remain in the batch queue. New events will append to the end of the queue. When flushed (timeout or full), batcher will group them to dispatch. If it sees a boundary of revision or projectId, it'll split into other groups. All dispatched in the correct order.